ra2ce/analysis/losses/multi_link_origin_closest_destination.py

```diff
@@ -119,48 +75,37 @@ def _save_gpkg_analysis(
                 )
         else:
             (
-                base_graph,
+                _base_graph,
                 origins,
                 destinations,
                 agg_results,
                 opt_routes_with_hazard,
             ) = analyzer.multi_link_origin_closest_destination()
             opt_routes_without_hazard = GeoDataFrame()
 
-        if self.analysis.save_gpkg:
-            # Save the GeoDataFrames
-            to_save_gdf = [
-                origins,
-                destinations,
-                opt_routes_without_hazard,
-                opt_routes_with_hazard,
+        _nodes_graph, _edges_graph = get_nodes_and_edges_from_origin_graph(_base_graph)
+        _base_name = self.analysis.name.replace(" ", "_")
+        _analysis_result_wrapper = AnalysisResultWrapper(
+            results_collection=[
+                self._get_analysis_result(origins, _base_name + "_origins"),
+                self._get_analysis_result(destinations, _base_name + "_destinations"),
+                self._get_analysis_result(destinations, _base_name + "_optimal_routes"),
+                self._get_analysis_result(_nodes_graph, _base_name + "_results_nodes"),
+                self._get_analysis_result(_edges_graph, _base_name + "_results_edges"),
             ]
-            to_save_gdf_names = [
-                "origins",
-                "destinations",
-                "optimal_routes_without_hazard",
-                "optimal_routes_with_hazard",
-            ]
-            _save_gpkg_analysis(base_graph, to_save_gdf, to_save_gdf_names)
-        if self.analysis.save_csv:
-            csv_path = _output_path.joinpath(
-                self.analysis.name.replace(" ", "_") + "_destinations.csv"
+        )
+        if not opt_routes_with_hazard.empty:
+            _analysis_result_wrapper.results_collection.append(
+                self._get_analysis_result(
+                    opt_routes_without_hazard, _base_name + "_optimal_routes"
+                )
             )
-            if "geometry" in destinations.columns:
-                del destinations["geometry"]
-            if not csv_path.parent.exists():
-                csv_path.parent.mkdir(parents=True)
-            destinations.to_csv(csv_path, index=False)
-
-            csv_path = _output_path.joinpath(
-                self.analysis.name.replace(" ", "_") + "_optimal_routes.csv"
+        if not opt_routes_without_hazard.empty:
+            _analysis_result_wrapper.results_collection.append(
+                self._get_analysis_result(
+                    opt_routes_with_hazard, _base_name + "_optimal_routes"
+                )
```

Comment:
Are these 2 branches mutually exclusive? If not, this might lead to overwriting a file (double use of "_optimal_routes". If so, an `elif` could make this explicit.

Reply:
This is as it was prior to my changes, either one or the other. In addition if I change the suffix (as I originally intended to) then the tests will fail as the generated files will not be as expected. 
I could, either leave it as it was (legacy approach) or just generate new files with `_optimal_routes_without_hazard` and `_optimal_routes_with_hazard` (as I did in an earlier commit).

Reply:
Keeping it as is is the safer option then, I'd say.